### PR TITLE
move cosmos implementation to its own file

### PR DESCRIPTION
**Pull request by tamarrow**

### CODE REVIEW

**Comment by jsancio** [comment]
@tamarrow rebase

**Comment by tamarrow** [comment]
@jsancio updated

File: cli/dcoscli/package/main.py [hunk 1]
```diff
@@ -718,3 +718,16 @@ def _bundle_screenshots(screenshot_directory, zip_file):
...
+    """Returns type of package manager to use
+
+    :returns: PackageManager instance
+    :rtype: PackageManager
+    """
+    url = os.environ.get(constants.COSMOS_URL_ENV)
+    if url:
+        return cosmospackage.Cosmos(url)
+    else:
+        return package.PackageManager()
```
Comment (jsancio) [inline cli/dcoscli/package/main.py]: :+1:

File: dcos/cosmospackage.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+
+            packages.append(result)
+
+        return packages
+
+
+class CosmosPackageVersion(package.PackageVersion):
+    """Interface to a specific package version from cosmos"""
+
+    def __init__(self, name, version, url):
```
Comment (jsancio) [inline dcos/cosmospackage.py]: I suspect that `version` is `releaseVersion`. Let's rename the property.

**Comment by jsancio** [comment]
LGTM. I have some questions around "source", "url" and "version" but we can address when we try to make the tests pass.

**Comment by jsancio** [comment]
:shipit:

**Pull request merged**